adafruit_bme280/basic.py

```diff
@@ -0,0 +1,379 @@
+# SPDX-FileCopyrightText: 2017 ladyada for Adafruit Industries
+#
+# SPDX-License-Identifier: MIT
+
+"""
+`adafruit_bme280`
+=========================================================================================
+
+CircuitPython driver from BME280 Temperature, Humidity and Barometric
+Pressure sensor
+
+* Author(s): ladyada
+
+Implementation Notes
+--------------------
+
+**Hardware:**
+
+* `Adafruit BME280 Temperature, Humidity and Barometric Pressure sensor
+  <https://www.adafruit.com/product/2652>`_ (Product ID: 2652)
+
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://circuitpython.org/downloads
+
+* Adafruit's Bus Device library:
+  https://github.com/adafruit/Adafruit_CircuitPython_BusDevice
+
+"""
+import math
+from time import sleep
+from micropython import const
+
+try:
+    import struct
+except ImportError:
+    import ustruct as struct
+
+
+__version__ = "2.6.4"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_BME280.git"
+
+#    I2C ADDRESS/BITS/SETTINGS
+#    -----------------------------------------------------------------------
+_BME280_ADDRESS = const(0x77)
+_BME280_CHIPID = const(0x60)
+
+_BME280_REGISTER_CHIPID = const(0xD0)
+_BME280_REGISTER_DIG_T1 = const(0x88)
+_BME280_REGISTER_DIG_H1 = const(0xA1)
+_BME280_REGISTER_DIG_H2 = const(0xE1)
+_BME280_REGISTER_DIG_H3 = const(0xE3)
+_BME280_REGISTER_DIG_H4 = const(0xE4)
+_BME280_REGISTER_DIG_H5 = const(0xE5)
+_BME280_REGISTER_DIG_H6 = const(0xE7)
+
+_BME280_REGISTER_SOFTRESET = const(0xE0)
+_BME280_REGISTER_CTRL_HUM = const(0xF2)
+_BME280_REGISTER_STATUS = const(0xF3)
+_BME280_REGISTER_CTRL_MEAS = const(0xF4)
+_BME280_REGISTER_CONFIG = const(0xF5)
+_BME280_REGISTER_PRESSUREDATA = const(0xF7)
+_BME280_REGISTER_TEMPDATA = const(0xFA)
+_BME280_REGISTER_HUMIDDATA = const(0xFD)
+
+_BME280_HUMIDITY_MIN = const(0)
+_BME280_HUMIDITY_MAX = const(100)
+
+"""iir_filter values"""
+IIR_FILTER_DISABLE = const(0)
+
+"""overscan values for temperature, pressure, and humidity"""
+OVERSCAN_DISABLE = const(0x00)
+OVERSCAN_X1 = const(0x01)
+OVERSCAN_X2 = const(0x02)
+OVERSCAN_X4 = const(0x03)
+OVERSCAN_X8 = const(0x04)
+OVERSCAN_X16 = const(0x05)
+
+_BME280_OVERSCANS = {
+    OVERSCAN_DISABLE: 0,
+    OVERSCAN_X1: 1,
+    OVERSCAN_X2: 2,
+    OVERSCAN_X4: 4,
+    OVERSCAN_X8: 8,
+    OVERSCAN_X16: 16,
+}
+
+"""mode values"""
+MODE_SLEEP = const(0x00)
+MODE_FORCE = const(0x01)
+MODE_NORMAL = const(0x03)
+
+_BME280_MODES = (MODE_SLEEP, MODE_FORCE, MODE_NORMAL)
```

Comment:
I'd probably move these to advanced. In the internal init just use the value. These cost memory for each variable name.